config/config.go

```diff
@@ -36,56 +40,82 @@ type RawConfig struct {
 // Each ChainConfig is defined as one ENV variable, where its content is JSON configuration for one chain/domain.
 // Variables are named like this: CBH_DOM_X where X is domain id.
 //
-func GetConfigFromENV() (Config, error) {
+func GetConfigFromENV(config *Config) (*Config, error) {
 	rawConfig, err := loadFromEnv()
 	if err != nil {
-		return Config{}, err
+		return config, err
 	}
 
-	return processRawConfig(rawConfig)
+	return processRawConfig(rawConfig, config)
 }
 
 // GetConfigFromFile reads config from file, validates it and parses
 // it into config suitable for application
-func GetConfigFromFile(path string) (Config, error) {
+func GetConfigFromFile(path string, config *Config) (*Config, error) {
 	rawConfig := RawConfig{}
 
 	viper.SetConfigFile(path)
 	viper.SetConfigType("json")
 
 	err := viper.ReadInConfig()
 	if err != nil {
-		return Config{}, err
+		return config, err
 	}
 
 	err = viper.Unmarshal(&rawConfig)
 	if err != nil {
-		return Config{}, err
+		return config, err
 	}
 
-	return processRawConfig(rawConfig)
+	return processRawConfig(rawConfig, config)
 }
 
-func processRawConfig(rawConfig RawConfig) (Config, error) {
-	config := Config{}
+// GetConfigFromNetwork fetches shared configuration from URL and parses it.
+func GetConfigFromNetwork(url string, config *Config) (*Config, error) {
```

Comment:
I would rename it then accordingly to something like GetChainConfigFromNetwork or GetSharedConfigFromNetwork